sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -533,7 +533,8 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
                   case u: UserDefinedType[_] =>
                     alter.failAnalysis(s"Cannot update ${table.name} field $fieldName type: " +
                       s"update a UserDefinedType[${u.sql}] by updating its fields")
-                  case _: CalendarIntervalType =>
+                  case _: CalendarIntervalType | _: YearMonthIntervalType |
+                       _: DayTimeIntervalType =>
```

Comment:
Hm, so we should fix all these kind of blacklisting .. while this fix is good, I wonder if we can sweep and fix in batch.

Reply:
Actually, I have found another places where `DayTimeINtervalType` and `YearMonthIntervalType` are not considered.
In terms of blacklisting, I found this part only (if I didn't miss something).
The rest of places we need to consider is as follows.

* https://github.com/apache/spark/blob/4f51e0045e2754035e304979182c0f583c65f174/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateUnsafeProjection.scala#L42
* https://github.com/apache/spark/blob/4f51e0045e2754035e304979182c0f583c65f174/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L155
* https://github.com/apache/spark/blob/4f51e0045e2754035e304979182c0f583c65f174/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L191
* https://github.com/apache/spark/blob/4f51e0045e2754035e304979182c0f583c65f174/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ArrayBasedMapBuilder.scala#L37
* https://github.com/apache/spark/blob/4f51e0045e2754035e304979182c0f583c65f174/sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcHadoopFsRelationSuite.scala#L40
* https://github.com/apache/spark/blob/master/sql/hive/src/test/scala/org/apache/spark/sql/sources/ParquetHadoopFsRelationSuite.scala#L41
* https://github.com/apache/spark/blob/4f51e0045e2754035e304979182c0f583c65f174/sql/hive/src/test/scala/org/apache/spark/sql/sources/JsonHadoopFsRelationSuite.scala#L35
* https://github.com/apache/spark/blob/4f51e0045e2754035e304979182c0f583c65f174/sql/hive/src/test/scala/org/apache/spark/sql/sources/SimpleTextHadoopFsRelationSuite.scala#L36

I think, we need to fix them with beyond just blacklisting and they need to be more carefully considered while this part can be simply fixed.